File: longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if s==" ":
            return 1
        s=list(s)
        temp=""
        maxlen=0
        length=0
        for i in range(len(s)):
            if s[i] not in temp:
                temp+=s[i]
                length=len(temp)
            else:
                temp=temp[temp.index(s[i])+1:]
                temp+=s[i]
                maxlen=max(maxlen,length)
        return max(maxlen,length)
        
        

File: longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int len=0,l=0;
        unordered_map<char,int> mpp;
        for(int r=0;r<s.size();r++){
            if(mpp.find(s[r])!=mpp.end()){
                l=max(mpp[s[r]]+1,l);
            }
                mpp[s[r]]=r;
               
    
            len=max(len,r-l+1);
        }
    return len;
    }
    
};